packages/lib/onedrive-api.ts

```diff
@@ -339,6 +341,15 @@ export default class OneDriveApi {
 
 					await handleRequestRepeat(error);
 					continue;
+				} else if (error?.code === 'activityLimitReached' && response?.headers?._headers['retry-after'][0]) {
+					// Wait for OneDrive throttling
+					// Relavent Microsoft Docs: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#best-practices-to-handle-throttling
+					// Decrement retry count as multiple sync threads will cause repeated throttling errors - this will wait until throttling is resolved to continue, preventing a hard stop on the sync
+					i--;
+					const sleep_seconds = response.headers._headers['retry-after'][0];
+					logger.info(`OneDrive Throttle, sync thread sleeping for ${sleep_seconds} seconds...`);
```

Comment:
Makes sense to at least log it. Do you know how long they make us wait in general? As you mentioned if it takes a very long time perhaps we should display this to the user, although I'm not sure how at the moment.

Reply:
I tested with a very aggressive 20 "max concurrent connections". 
It took about 15min before the bulk download could resume. 

The typical 'retry-count' header maxes around 5min, but with 20 threads it seemed to drag on. Essentially, you end up with 20 threads retrying every `retry-count HTTP header` seconds. If it fails due to throttling, that request still counts against your request quota and increases the duration of throttling. 

Less threads will not have such large wait times, I would imagine.

Unfortunately, as far as I can tell, Microsoft doesn't explicitly say what the limits are, so it's hard to say for certian.

As pointed out by another contributor, the batch endpoint would be a great enhancement, however, it would take a bit of a refactor.